v3/src/models/data/formula-fn-registry.ts

```diff
@@ -44,6 +44,50 @@ const aggregateFnWithFilterFactory = (fn: (values: number[]) => number) => {
 // count(attribute) will return a count of valid data values, since 0 is a valid numeric value.
 export const isValueTruthy = (value: any) => value !== "" && value !== false && value !== null && value !== undefined
 
+
+// `next` and `prev` functions can share the same implementation assuming that `next` is just reversed `prev`.
+// `next` function needs to be executed for each case in the reverse order, and it'll also have different case pointer
+// modifier. `prev` will set scope to the previous index (-1), while next to the next one (+1).
+export const prevOrNextFactory = (variant: "next" | "prev") =>
+  (args: MathNode[], mathjs: any, scope: FormulaMathJsScope) => {
+  interface ICachedData {
+    result?: FValue
+  }
+
+  const cacheKey = `${variant}(${args.toString()})-${scope.getSemiAggregateGroupId()}`
+  const [ expression, defaultValue, filter ] = args
+  const cachedData = scope.getCached(cacheKey) as ICachedData | undefined
+  let result
+
+  if (cachedData !== undefined) {
+    // In case we don't find a new result, we need to reuse the old one.
+    result = cachedData.result
+    let newExpressionValue, newFilterValue
+    // This block will resolve attribute names to previous case values.
```

Comment:
```suggestion
    // This block will resolve attribute names to previous/next case values.
```

Reply:
This code is gone in the newest version, so I'm not committing this change to avoid merge conflicts.